File: birdstagram/spec/system/user_can_sign_up_spec.rb
require 'rails_helper'

RSpec.describe "User can sign up", type: :system do
  before do
    driven_by(:rack_test)
  end

  it 'shows the user a page for their username and password' do
    visit '/users/new'
    fill_in 'user_username', with: 'rio'
    fill_in 'user_password', with: 'bells'
    fill_in 'user_email', with: 'rio@example.com'
    click_button 'Sign Up'
    expect(page).to have_content ('Hello, rio')
    user_instance = User.order("id").last

    expect(user_instance.username).to eq('rio')
    expect(user_instance.email).to eq('rio@example.com')
  end
end

File: cypress/integration/user_can_sign_up_spec.js
describe("Sign up", () => {
  it("allows users to sign up with their details", () => {
    cy.visit("/users/new");
    cy.get("#username").type("person1");
    cy.get("#email").type("person1@email.co.uk");
    cy.get("#password").type("password");
    cy.get("#submit").click();

    cy.url().should("eq", "http://localhost:3030/")
  });
});